dashboard/lib/services/lti.rb

```diff
@@ -27,6 +28,13 @@ def self.initialize_lti_user(id_token)
     user
   end
 
+  def self.create_lti_user_identity(user)
+    auth_option = user.authentication_options.find {|ao| ao.credential_type == AuthenticationOption::LTI_V1}
+    issuer, client_id, subject = auth_option.authentication_id.split('|')
+    lti_integration = Queries::Lti.get_lti_integration(issuer, client_id)
+    LtiUserIdentity.create(user: user, subject: subject, lti_integration: lti_integration)
```

Comment:
In the `initialize_lti_user` function we're doing a `new` instead of a `create`, and returning that user at the end. Do we want to do the same here and return a `new` LtiUserIdentity, then call `save` on it in the after_create hook?

Reply:
The reason we are calling `new` in the above method is because we are actually caching the user attributes, and then reading those attributes back out when we take the user to the complete signup flow. This is because we'll get validation errors if we don't have an email (for teachers) and age (for students), and we need them to supply those in on the signup page. Since we don't need to do that here, it seems most simple to just create the lti_user_identity, since it's being invoked as a after_create callback. Thoughts?

Reply:
I think that's reasonable, especially given the fact that the function has "create" in the name, implying the create method will be called on the object.